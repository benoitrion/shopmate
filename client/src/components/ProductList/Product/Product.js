import React from 'react';

import Card, {
  CardPrimaryContent,
  CardMedia,
  CardActions,
  CardActionButtons
} from '@material/react-card/'

import { Button } from '@material/react-button';
import MaterialIcon from '@material/react-material-icon';

import {
  Headline6,
  Body2
} from '@material/react-typography/'

import '@material/react-card/index.scss';
import '@material/react-typography/index.scss';
import '@material/react-button/index.scss';
import '@material/react-material-icon/index.scss';

import './Product.scss';

function Product({ product }) {
  const image = require(`../../../static/product_images/${product.image}`);
  return (
    <React.Fragment>
      <Card className='product-card'>
        <CardPrimaryContent>
          <CardMedia square imageUrl={image} alt={product.name} className='product-card__media' />
        </CardPrimaryContent>
        <div className='product-card__primary'>
          <Headline6 className='product-card__title'>
            {product.name}
          </Headline6>
        </div>
        <Body2 className='product-card__body'>
          {product.description}
        </Body2>
        <CardActions>
          <CardActionButtons>
            <Button className='product-card__button'>
              <MaterialIcon icon='add_shopping_cart' className='product-card__icon' />
              <span className="mdc-button__label">Add to cart</span>
            </Button>
          </CardActionButtons>
        </CardActions>
      </Card>
    </React.Fragment>
  );
}

export default Product;
