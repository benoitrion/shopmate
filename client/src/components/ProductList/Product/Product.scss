.product-card {
}
.product-card__media {
}
.product-card__primary {
}
.product-card__title {
  margin: 10px;
}
.product-card__body {
  height: 100px;
  margin: 10px;
}
.product-card__icon {
  margin-right: 5px;
  display: inline-block;
  vertical-align:middle;
}