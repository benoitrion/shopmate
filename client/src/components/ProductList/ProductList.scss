.product-list_grid {
  margin: 70px auto 0 auto;
  width: 1200px;
}