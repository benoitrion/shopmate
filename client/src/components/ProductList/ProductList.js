import React from 'react';
import PropTypes from 'prop-types';

import { Cell, Grid, Row } from '@material/react-layout-grid';

import '@material/react-layout-grid/index.scss';

import Product from './Product/Product';

import './ProductList.scss';

function ProductList({ products }) {
  return (
    <React.Fragment>
      <Grid className="product-list_grid">
        <Row>
          {products.map(p => {
            return (
              <Cell columns={3} key={p.id}>
                  <Product product={p}/>
              </Cell>
            )
          })}
        </Row>
      </Grid>
    </React.Fragment>
  )
}

ProductList.propTypes = {
  products: PropTypes.array.isRequired
};

export default ProductList;
