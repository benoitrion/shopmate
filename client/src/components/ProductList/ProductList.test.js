import React from 'react';
import { shallow } from 'enzyme';
import ProductList from './ProductList';
import Product from './Product/Product';

describe('ProductList test', () => {
  const mockProducts = [
    {
      id:1,
      name: 'Arc d\'Triomphe',
      description: 'This beautiful and iconic T-shirt will no doubt lead you to your own triumph.',
      price: 14.99,
      discounted_price: 0.00,
      image: 'arc-d-triomphe.gif',
      image_2: 'arc-d-triomphe-2.gif',
      thumbnail: 'arc-d-triomphe-thumbnail.gif',
      display: 0
    },
    {
      id:2,
      name: 'name2',
      description: 'This beautiful and iconic T-shirt will no doubt lead you to your own triumph.',
      price: 14.99,
      discounted_price: 0.00,
      image: 'arc-d-triomphe.gif',
      image_2: 'arc-d-triomphe-2.gif',
      thumbnail: 'arc-d-triomphe-thumbnail.gif',
      display: 0
    }
  ];
  let wrapper;
  beforeEach(() => wrapper = shallow(<ProductList products={mockProducts}/>));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());
  
  it('renders two <Product /> components', () => {
    expect(wrapper.find(Product).length).toEqual(2);
  });

});