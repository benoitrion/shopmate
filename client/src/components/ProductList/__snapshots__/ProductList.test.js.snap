// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList test should render correctly 1`] = `
<Fragment>
  <Grid
    className="product-list_grid"
  >
    <Row>
      <Cell
        columns={4}
        key="1"
      >
        <Product
          product={
            Object {
              "description": "This beautiful and iconic T-shirt will no doubt lead you to your own triumph.",
              "discounted_price": 0,
              "display": 0,
              "id": 1,
              "image": "arc-d-triomphe.gif",
              "image_2": "arc-d-triomphe-2.gif",
              "name": "Arc d'Triomphe",
              "price": 14.99,
              "thumbnail": "arc-d-triomphe-thumbnail.gif",
            }
          }
        />
      </Cell>
      <Cell
        columns={4}
        key="2"
      >
        <Product
          product={
            Object {
              "description": "This beautiful and iconic T-shirt will no doubt lead you to your own triumph.",
              "discounted_price": 0,
              "display": 0,
              "id": 2,
              "image": "arc-d-triomphe.gif",
              "image_2": "arc-d-triomphe-2.gif",
              "name": "name2",
              "price": 14.99,
              "thumbnail": "arc-d-triomphe-thumbnail.gif",
            }
          }
        />
      </Cell>
    </Row>
  </Grid>
</Fragment>
`;
